site/hxapi/asynchronous-bookings.md

```diff
@@ -0,0 +1,143 @@
+
+Partners must include a new parameter in the availability request to indicate whether to include async products.  
+
+| Name                    | Data Type | Format  | Mandatory? | Additional Information                                    |
+|-------------------------|-----------|---------|------------|----------------------------------------------------------|
+| includeAsyncProducts    | Boolean   | `true`/`false` | N | If `true`, availability includes async products. Default is `false`. |
+
+### UK Products Availability Request
+
+{% codetabs %}
+{% codetab XML %}
+```
+https://api.holidayextras.co.uk/carpark/LGW?ABTANumber=YourABTANumber&Password=YourPassword&Initials=YourInitials&key=YourKey&token=YourToken&ArrivalDate=2020-01-07&ArrivalTime=1200&DepartDate=2020-01-08&DepartTime=1200&NumberOfPax=1&includeAsyncProducts=true
+```
+{% endcodetab %}
+{% codetab JSON %}
+```
+https://api.holidayextras.co.uk/carpark/LGW.js?ABTANumber=YourABTANumber&Password=YourPassword&Initials=YourInitials&key=YourKey&token=YourToken&ArrivalDate=2020-01-07&ArrivalTime=1200&DepartDate=2020-01-08&DepartTime=1200&NumberOfPax=1&includeAsyncProducts=true
+```
+Note the `.js` extension.
+{% endcodetab %}
+{% endcodetabs %}
+
+### European Products Availability Request
+
+{% codetabs %}
+{% codetab XML %}
+```
+https://api.holidayextras.co.uk/carpark/MUC?ABTANumber=YourABTANumber&Password=YourPassword&Initials=YourInitials&key=YourKey&token=YourToken&ArrivalDate=2017-12-01&ArrivalTime=1200&DepartDate=2017-12-08&DepartTime=1200&NumberOfPax=1&System=ABG&lang=en&includeAsyncProducts=true
+```
+{% endcodetab %}
+{% codetab JSON %}
+```
+https://api.holidayextras.co.uk/carpark/MUC.js?ABTANumber=YourABTANumber&Password=YourPassword&Initials=YourInitials&key=YourKey&token=YourToken&ArrivalDate=2017-12-01&ArrivalTime=1200&DepartDate=2017-12-08&DepartTime=1200&NumberOfPax=1&System=ABG&lang=en&includeAsyncProducts=true
+```
+Note the `.js` extension.
+{% endcodetab %}
+{% endcodetabs %}
+
+
+### Behavior  
+
+- **`true`:** Availability searches include async products.  
+- **`false`:** Availability searches exclude async products (default behavior).  
+
+---
+
+## Booking Status Polling  
+
+When working with asynchronous bookings, partners must poll our API to determine the status of a booking and when full booking details are ready.  
+
+### When Polling is Required  
+
+Polling is required when a booking has been submitted but is not yet confirmed. The `bookingStatus` field in the response from the `view booking` endpoint indicates the current status of a booking.  
+
+Booking details such as the QR code, supplier reference, and other information will only be provided once the `bookingStatus` field is in a `CONFIRMED` state.
+
+### How Often to Poll  
+
+The frequency of polling depends on the product type and supplier:  
+- **Standard Products:** Poll every 5–10 minutes.  
+- **Long Confirmation Times (e.g., specific suppliers):** Poll every 15–30 minutes.  
```

Comment:
How would a partner reading this know which products are which?

I know we discussed making these more dynamic (i.e. we'd tell them how often to poll in the booking response), but don't remember what we decided.

Reply:
How would a partner know which products are async and which products are not?

Good question I don't think we discussed about this.

Reply:
This ticket was done https://holidayextras.jira.com/browse/PRTNR-1013 so now with that flag sent async booking products will be returned on the response.